src/modules/statusUpdates.js

```diff
@@ -0,0 +1,33 @@
+import UI from './ui';
+
+const ul = document.querySelector('.ul');
+
+const handleStatusUpdate = () => {
+  ul.addEventListener('change', (e) => {
+    if (e.target.classList.contains('check')) {
+      const taskId = e.target.id;
+      let todoData = UI.getItem();
+
+      // Filter out the task with the specified taskId and update its completed status
+      todoData = todoData.filter((todo) => {
+        if (todo.index === taskId) {
+          todo.completed = e.target.checked;
+        }
+        return todo;
+      });
+
+      // Update the task in the localStorage
+      localStorage.setItem('todoData', JSON.stringify(todoData));
+
+      // Update the completed class on the input element
+      const input = e.target.nextElementSibling;
+      if (e.target.checked) {
+        input.classList.add('completed');
+      } else {
+        input.classList.remove('completed');
+      }
+    }
+  });
+};
```

Comment:
### Please make your code more DRY by destructuring the event object. 
- Destructuring allows you to extract properties from objects in a concise and readable way. 
- Destructuring eliminates the needs to repeats `e.target.checked`.
### I recommend you modify your code like this: 
```javascript
const handleStatusUpdate = () => {
  ul.addEventListener('change', (e) => {
  const { target } = e
  const { classList, checked, id, nextElementSibling } = target
    if (classList.contains('check')) {
      let todoData = UI.getItem();

      // Filter out the task with the specified taskId and update its completed status
      todoData = todoData.filter((todo) => {
        if (todo.index === id) {
          todo.completed = checked;
        }
        return todo;
      });

      // Update the task in the localStorage
      localStorage.setItem('todoData', JSON.stringify(todoData));

      // Update the completed class on the input element
      const input = nextElementSibling;
      if (checked) {
        input.classList.add('completed');
      } else {
        input.classList.remove('completed');
      }
    }
  });
};